src/usb_stream.cpp

```diff
@@ -420,6 +420,10 @@ bool UsbStream::read_device(std::vector<uint8_t>& msg, const bool stream_on)
     else
     {
         return_value_ = read(file_id_, buffer_.data(), msg.size());
+        // try to read again if bytes requested != bytes received 
```

Comment:
shouldn't it be a while loop here? with a timeout?

Reply:
Perhaps. I wanted to make as small of a change as possible, and just 1 retry is enough to get the IMU working.

Reply:
Do you want me to change it to a while loop?

Reply:
Maybe not the while loop but at least manage the potential error:
what happens if after 2 trial you still do not have the messages you need?

Reply:
The original code didn't have this "rereading" function at all. I can do a loop that gives it 3 tries at reading the correct amount of data. Would that be OK?

Reply:
Let's have a look at this in the lab later.